src/ui/public/saved_objects/saved_object.js

```diff
@@ -0,0 +1,35 @@
+import _ from 'lodash';
+
+export class SavedObject {
+  constructor(client, { id, type, version, attributes } = {}) {
+    this._client = client;
+    this.id = id;
+    this.type = type;
+    this._attributes = attributes || {};
+    this._version = version;
+  }
+
+  get(key) {
+    return _.get(this._attributes, key);
+  }
+
+  set(key, value) {
+    return _.set(this._attributes, key, value);
+  }
+
+  has(key) {
+    return _.has(this._attributes, key);
+  }
+
+  save() {
+    if (this.id) {
+      return this._client.update(this.type, this.id, this._attributes);
```

Comment:
There will be times that we want to create a SavedObject with a specific id, like for index patterns. The current implementation of the PUT route requires that the document already exist.